packages/web/src/components/search/DataSearch.js

```diff
@@ -80,8 +86,11 @@ class DataSearch extends Component {
 		props.setQueryListener(props.componentId, props.onQueryChange, props.onError);
 
 		if (props.highlight) {
-			const queryOptions = DataSearch.highlightQuery(props) || {};
+			let queryOptions = DataSearch.highlightQuery(props) || {};
 			queryOptions.size = 20;
+			if (this.shouldGenerateAggsQuery) {
```

Comment:
You're repeating this block, it's better to create a method to update the query options based on the `aggregationField`.